fetch/fetch.go

```diff
@@ -588,7 +599,7 @@ func (f *Fetch) send(requests []RequestMessage) {
 			go func() {
 				data, err := f.sendBatch(peer, batch)
```

Comment:
Not part of this PR, but this part of the code seems like it can be improved:

- the spawned go routine isn't tracked via an `errgroup.Group` or `sync.Waitgroup`
- there is no cancellation via a `context.Context` in place, the caller has no control over when sending a request shall be aborted (shutdown or timeout)
- `handleHashError` locks a mutex for its whole lifetime, while `receiveResponse` does not. So if something is wrong and a lot of errors happen everything might slow down 🤔